fml/icu_util.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/icu_util.h"
+#include <memory>
+#include <mutex>
+#include "flutter/fml/file_mapping.h"
+#include "flutter/fml/resource_mapping.h"
+#include "lib/ftl/logging.h"
+#include "third_party/icu/source/common/unicode/udata.h"
+
+namespace fml {
+namespace icu {
+
+static const char kIcuDataFileName[] = "icudtl.dat";
+
+class ICUContext {
+ public:
+  ICUContext(const std::string& icu_data_path)
+      : valid_(SetupMapping(icu_data_path) && SetupICU()) {}
+
+  ~ICUContext() = default;
+
+  bool SetupMapping(const std::string& icu_data_path) {
+    // Check if the explicit path specified exists.
+    auto overriden_path_mapping = std::make_unique<FileMapping>(icu_data_path);
+    if (overriden_path_mapping != nullptr &&
```

Comment:
It can't I think just repeated the resource mapping logic. Will get rid of this and the other one for file mapping.